plan/column_pruning.go

```diff
@@ -157,16 +157,21 @@ func (p *Union) PruneColumns(parentUsedCols []*expression.Column) {
 }
 
 // PruneColumns implements LogicalPlan interface.
+func (p *LogicalUnionScan) PruneColumns(parentUsedCols []*expression.Column) {
+	for _, col := range p.schema.TblID2Handle {
+		parentUsedCols = append(parentUsedCols, col[0])
+	}
+	p.children[0].(LogicalPlan).PruneColumns(parentUsedCols)
+	p.SetSchema(p.children[0].Schema())
```

Comment:
if `PkIsHandle` is false, `p.Schema` can remove the last col of p.children[0].Schema() ?

Reply:
the following project can prove there is no extra column.

Reply:
If set a proper schema for UnionScan,
the Prj may be eliminated.

Reply:
The length of union scan’ schema is equal to the length of row it outputs. The projection can’t be eliminate.